File: Go/pkg/sort/selection_sort.go
package sort

func selectionSort(valuesToSort []int) []int {
	n := len(valuesToSort)

	for i := 0; i < n-1; i++ {
		minimum_index := i
		for j := i + 1; j < n; j++ {
			if valuesToSort[j] < valuesToSort[minimum_index] {
				minimum_index = j
			}
		}

		temp := valuesToSort[minimum_index]
		valuesToSort[minimum_index] = valuesToSort[i]
		valuesToSort[i] = temp
	}
	return valuesToSort
}

File: Rust/algorithms/src/selection_sort.rs
pub fn sort(values_to_sort: &mut [i32]) {
    let n = values_to_sort.len();
    let mut i = 0;

    while i < n - 1 {
        let mut minimum_index = i;
        let mut j = i;
        while j < n {
            if values_to_sort[j] < values_to_sort[minimum_index] {
                minimum_index = j;
            }
            j += 1;
        }
        let temp = values_to_sort[minimum_index];
        values_to_sort[minimum_index] = values_to_sort[i];
        values_to_sort[i] = temp;
        i += 1;
    }
}
